deploy/custom_resources/custom_authorizer/jwt_services.py

```diff
@@ -53,49 +20,55 @@ def fetch_public_keys():
     'verify_iss': True,
     'verify_sub': True,
     'verify_jti': True,
-    'verify_at_hash': False,
-    'require_aud': True,
-    'require_iat': True,
-    'require_exp': True,
-    'require_nbf': False,
-    'require_iss': True,
-    'require_sub': True,
-    'require_jti': True,
-    'require_at_hash': False,
-    'leeway': 0,
+    'verify_at_hash': True,
+    'require': ['aud', 'iat', 'exp', 'iss', 'sub', 'jti'],  # "nbf", "at_hash"
 }
 
 
 class JWTServices:
+    @staticmethod
+    def _fetch_openid_url(key):
+        response = requests.get(f'{os.environ.get("custom_auth_url")}/.well-known/openid-configuration')
+        response.raise_for_status()
+        return response.json().get(key)
+
     @staticmethod
     def validate_jwt_token(jwt_token):
         try:
-            # Decode and verify the JWT token
-            header = get_unverified_header(jwt_token)
-            kid = header['kid']
-            if kid not in issuer_keys:
-                logger.info('Public key not found in provided set of keys')
-                # Retry Fetching the public certificates again in case rotation occurs and lambda has cached the publicKeys
-                fetch_public_keys()
-                if kid not in issuer_keys:
-                    raise Exception('Unauthorized')
-            public_key = issuer_keys.get(kid)
-            payload = decode(
+            # get JWK URI from OpenId Configuration
+            jwks_url = JWTServices._fetch_openid_url('jwks_uri')
+
+            # Init pyJWT.JWKClient with JWK URI
+            jwks_client = jwt.PyJWKClient(jwks_url)
+
+            # get signing_key from JWT
+            signing_key = jwks_client.get_signing_key_from_jwt(jwt_token)
+
+            # Decode and Verify JWT
+            payload = jwt.decode(
                 jwt_token,
-                public_key.get('jwk'),
+                signing_key.key,
                 algorithms=['RS256', 'HS256'],
-                issuer=public_key.get('issuer'),
-                audience=public_key.get('audience'),
+                issuer=os.environ.get('custom_auth_url'),
+                audience=os.environ.get('custom_auth_client'),
+                leeway=0,
                 options=jwt_options,
             )
-
             return payload
-        except ExpiredSignatureError:
+        except jwt.exceptions.ExpiredSignatureError:
```

Comment:
nit: I would directly raise these Exceptions here instead of passing the None and then raising the general exception